addons/stock/models/stock_move.py

```diff
@@ -0,0 +1,1109 @@
+        for move in self:
+            # if the move is preceeded, then it's waiting (if preceeding move is done, then action_assign has been called already and its state is already available)
+            if move.move_orig_ids:
+                move_waiting |= move
+            # if the move is split and some of the ancestor was preceeded, then it's waiting as well
+            else:
+                inner_move = move.split_from
+                while inner_move:
+                    if inner_move.move_orig_ids:
+                        move_waiting |= move
+                        break
+                    inner_move = inner_move.split_from
+                else:
+                    if move.procure_method == 'make_to_order':
+                        move_create_proc |= move
+                    else:
+                        move_to_confirm |= move
+
+            if not move.picking_id and move.picking_type_id:
+                key = (move.group_id.id, move.location_id.id, move.location_dest_id.id)
+                if key not in to_assign:
+                    to_assign[key] = self.env['stock.move']
+                to_assign[key] |= move
+
+        # create procurements for make to order moves
+        procurements = self.env['procurement.order']
+        for move in move_create_proc:
+            procurements |= procurements.create(move._prepare_procurement_from_move())
+        procurements.run()
+
+        move_to_confirm.write({'state': 'confirmed'})
+        (move_waiting | move_create_proc).write({'state': 'waiting'})
+
+        # assign picking in batch for all confirmed move that share the same details
+        for key, moves in to_assign.items():
+            moves.assign_picking()
+        self._push_apply()
+        return True
+
+    def _prepare_procurement_from_move(self):
+        origin = (self.group_id and (self.group_id.name + ":") or "") + (self.rule_id and self.rule_id.name or self.origin or self.picking_id.name or "/")
+        group_id = self.group_id and self.group_id.id or False
+        if self.rule_id:
+            if self.rule_id.group_propagation_option == 'fixed' and self.rule_id.group_id:
+                group_id = self.rule_id.group_id.id
+            elif self.rule_id.group_propagation_option == 'none':
+                group_id = False
+        return {
+            'name': self.rule_id and self.rule_id.name or "/",
+            'origin': origin,
+            'company_id': self.company_id.id,
+            'date_planned': self.date,
+            'product_id': self.product_id.id,
+            'product_qty': self.product_uom_qty,
+            'product_uom': self.product_uom.id,
+            'location_id': self.location_id.id,
+            'move_dest_id': self.id,
+            'group_id': group_id,
+            'route_ids': [(4, x.id) for x in self.route_ids],
+            'warehouse_id': self.warehouse_id.id or (self.picking_type_id and self.picking_type_id.warehouse_id.id or False),
```

Comment:
May be 'warehouse_id': self.warehouse_id.id or self.picking_type_id.warehouse_id.id .